File: nx/apps/wordcounter/wordcounter.js
#!/usr/bin/env node
/**
 * Word counter application - displays word dictionary statistics.
 * Similar to the UV wordcounter app, but implemented in Node.js for Nx.
 */

const fs = require("fs");
const path = require("path");

// Dictionary mapping folder names to their word lists
const WORD_DICT = {};

// Word list packages (similar to UV setup)
const wordPackages = ["alpha", "bravo", "charlie", "delta", "echo", "foxtrot", "golf"];

// Load words from each package's .txt file
for (const pkg of wordPackages) {
  const txtPath = path.join(__dirname, "../../", pkg, `${pkg}.txt`);
  try {
    const content = fs.readFileSync(txtPath, "utf-8");
    const words = content
      .split("\n")
      .map((line) => line.trim())
      .filter((word) => word.length > 0);
    WORD_DICT[pkg] = words;
  } catch (error) {
    console.error("Error loading", pkg + ":", error.message);
    WORD_DICT[pkg] = [];
  }
}

function main() {
  console.log("Nx Word Dictionary");
  console.log("=".repeat(50));

  for (const [folder, words] of Object.entries(WORD_DICT)) {
    console.log(`${folder}: ${words.length} words`);
  }

  const totalWords = Object.values(WORD_DICT).reduce((sum, words) => sum + words.length, 0);
  console.log(`\nTotal words: ${totalWords}`);
}

if (require.main === module) {
  main();
}

module.exports = { WORD_DICT, main };

File: uv/apps/wordcounter/wordcounter.py
"""Word counter application - displays word dictionary statistics."""

# Import from workspace packages in lib (uv-alpha, uv-bravo, etc.)
# Each package exposes its module directly
import alpha
import bravo
import charlie
import delta
import echo
import foxtrot
import golf

# Dictionary mapping folder names to their word lists
WORD_DICT = {
    "alpha": alpha.WORDS,
    "bravo": bravo.WORDS,
    "charlie": charlie.WORDS,
    "delta": delta.WORDS,
    "echo": echo.WORDS,
    "foxtrot": foxtrot.WORDS,
    "golf": golf.WORDS,
}

__all__ = ["WORD_DICT", "main"]


def main():
    """Main entry point - displays word dictionary statistics."""
    print("UV Word Dictionary")
    print("=" * 50)

    for folder, words in WORD_DICT.items():
        print(f"{folder}: {len(words)} words")

    print("\nTotal words:", sum(len(words) for words in WORD_DICT.values()))


if __name__ == "__main__":
    main()
